utils/criu_linux.go

```diff
@@ -0,0 +1,45 @@
+//go:build linux
+// +build linux
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/checkpoint-restore/go-criu/v7"
+	"github.com/checkpoint-restore/go-criu/v7/rpc"
+	"google.golang.org/protobuf/proto"
+)
+
+// CheckForCRIU checks if CRIU is available and if it is as least the
+// version as specified in the "version" parameter.
+func CheckForCriu(version int) error {
+	criuVersion, err := GetCriuVersion()
+	if err != nil {
+		return fmt.Errorf("failed to check for criu version: %w", err)
+	}
+
+	if criuVersion >= version {
+		return nil
+	}
+	return fmt.Errorf("checkpoint/restore requires at least CRIU %d, current version is %d", version, criuVersion)
+}
+
+// Convenience function to easily check if memory tracking is supported.
+func MemTrack() bool {
```

Comment:
Would it make more sense to name this `IsMemTrack()`?

Reply:
Sure. Will do.

Reply:
I changed it to `GetMemTrack()`. But I am also open to `IsMemTrack()`. Both approaches seem valid. I used `Get..` as that is used in one of the functions.

Reply:
The way I see it, `Get` signifies the actual object - in this case, possibly a handler or a struct that can work with memory tracking. `Is` signifies a purely existential check, which seems more appropriate in this context.

Reply:
Changed to `IsMemTrack()`